golang/v2/test/api_test.go

```diff
@@ -50,3 +50,78 @@ func TestAlerts(t *testing.T) {
 		t.Fatalf("ListAlerts() = %v, want 10", len(resp.Alerts))
 	}
 }
+
+//Testing ListAlerts with specific Zone passed. Return only those zone specific alerts
+func TestAlertsZoneFilter(t *testing.T) {
+	setup(t)
+
+	zone := "nightlycommon"
+	tenantID := "meta-tap"
+	filter := araali_api_service.AlertFilter{
+		Time: &araali_api_service.TimeSlice{
+			StartTime: timestamppb.New(time.Date(1980, time.November, 0, 0, 0, 0, 0, time.UTC)),
+			EndTime:   timestamppb.New(time.Now()),
+		},
+		ListAllAlerts:        false,
+		OpenAlerts:           true,
+		ClosedAlerts:         false,
+		PerimeterIngress:     true,
+		PerimeterEgress:      true,
+		HomeNonAraaliIngress: true,
+		HomeNonAraaliEgress:  true,
+		AraaliToAraali:       true,
+		Zone:                 zone,
+	}
+	resp, err := api.ListAlerts(tenantID, &filter, 10, "")
+	totalAlerts := len(resp.Alerts)
+
+	if verbose > 0 {
+		fmt.Printf("\nR: %+v/%v\n", totalAlerts, err)
+	}
+
+	counter := 0
+	for _, s := range resp.Alerts {
+		_, araaliEndpoint := s.Client.Info.(*araali_api_service.EndPoint_Araali)
+		if araaliEndpoint {
+			if s.Client.GetAraali().Zone == zone {
+				counter++
+			}
+		}
+	}
+
+	//Validate whether all the alerts are corresponding to passed zone value
+	if counter != totalAlerts {
+		t.Fatalf("%v count in the response = %v, want %v", zone, counter, totalAlerts)
+	}
+
+}
+
+//Testing ListAlerts with specific Zone passed which returns ZERO count
+func TestAlertsZoneFilterNoResults(t *testing.T) {
+	setup(t)
+
+	zone := "intern-nightly"
```

Comment:
Use an invalid zone that is guaranteed to have no alerts?

Reply:
Updated the zone name which is not there in the system("invalidzone")